app/views/cloud_subnet/new.html.haml

```diff
@@ -11,10 +11,10 @@
                       options_for_select([["<#{_('Choose')}>", nil]] + @network_provider_choices.sort),
```

Comment:
@gildub I also recommend using -
```ruby
options_for_select([["<#{_('Choose')}>", nil]] + @network_provider_choices.sort, disabled: ["<#{_('Choose')}>", nil]),
```
So that the user is forced to make a valid selection and cannot choose `<Choose>`
This change alone should help address the BZ, and you probably don't even need the other changes.

Reply:
@AparnaKarve, I was trying to align with the code on master branch but you're right that's better. Thanks